src/pipeline.h

```diff
@@ -201,6 +201,8 @@ struct PipelineBaton {
   std::string withIccProfile;
   std::unordered_map<std::string, std::string> withExif;
   bool withExifMerge;
+  void *withXmpBuffer;
```

Comment:
nit: You can use a `unique_ptr<uint8_t[]>` here, this would allow to let c++ handle the deletion on "error" cases (remove the handling in OnOK) and you have a [release](https://en.cppreference.com/w/cpp/memory/unique_ptr/release.html) function on it to pass the memory to "legacy" apis like VipsImage#set